src/librustc_mir/transform/const_prop.rs

```diff
@@ -997,8 +1021,53 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
             | TerminatorKind::GeneratorDrop
             | TerminatorKind::FalseEdges { .. }
             | TerminatorKind::FalseUnwind { .. } => {}
-            //FIXME(wesleywiser) Call does have Operands that could be const-propagated
-            TerminatorKind::Call { .. } => {}
+            TerminatorKind::Call { ref mut args, .. } => {
+                // Every argument in our function calls can be const propagated.
+                // We try to, before exiting.
+                for opr in args {
```

Comment:
Could we break these changes out into one more commit? (Sorry :p) 

I'd like to have one commit just be the "propagate locals within their own block" change and one commit be the "do const prop on function args" change.

Reply:
Of course!

Reply:
Done as well ^^